src/utils/response/responseHandlers.ts

```diff
@@ -6,20 +6,82 @@ import {
   ApiResponseOnFailure,
   ApiResponseOnFailureServerCommunication,
   ApiResponseOnSuccess,
+  ApiResponseBatchOnSuccess,
   Entity,
   LibraryReturn,
   ResourceNames,
+  ApiBatchResponse,
+  ApiResponseBatchOnFailure,
 } from './apiResponse';
 
 export type RequestPromise<ResourceName extends ResourceNames> = Promise<AxiosResponse<ApiResponse<ResourceName>>>;
 
+export type RequestBatchPromise<ResourceName extends ResourceNames> = Promise<AxiosResponse<ApiBatchResponse<ResourceName>>>;
+
 export type Required<ResourceName extends ResourceNames, Resource = Entity<ResourceName>> = Promise<LibraryReturn<ResourceName, Resource>>;
 
-export async function toApiResponse<ResourceName extends ResourceNames>(
-  promise: RequestPromise<ResourceName>
-): Promise<ApiResponseOnSuccess<ResourceName>> {
-  let resolved: AxiosResponse<ApiResponse<ResourceName>> | undefined;
+export type ParsedErrorMesage = {
+  response: ApiResponseOnFailure;
+  message: string;
+  stack: string | undefined;
+  code?: number;
+};
+
+type ParserErrorMesageObject<ResourceName extends ResourceNames> = ParsedErrorMesage & {
+  _plainError: AxiosResponse<ApiBatchResponse<ResourceName> | ApiResponse<ResourceName>>;
+  code: number;
+};
+
+export function getParsedErrorMessage(apiResponse: ApiResponseOnFailure): ParsedErrorMesage {
+  let errorMessage = 'Something went wrong';
+
+  if (typeof apiResponse.ErrorExtended?.errorString == 'string' && 0 < apiResponse.ErrorExtended.errorString.length) {
+    errorMessage = apiResponse.ErrorExtended.errorString;
+  } else if (typeof apiResponse.ErrorMessage == 'string') {
+    errorMessage = apiResponse.ErrorMessage;
+  }
+
+  return {
+    response: apiResponse,
+    message: errorMessage,
+    stack: new Error().stack,
+    code: apiResponse.Error,
+  };
+}
+
+function getParsedErrorMessageObject<ResourceName extends ResourceNames>(
+  apiResponse: ApiResponseOnFailure,
+  resolved: AxiosResponse<ApiBatchResponse<ResourceName>>
+): ParserErrorMesageObject<ResourceName>;
```

Comment:
let's maybe add some spaces between the overloads, at least here on GitHub they are a bit hard to read